torchprime/torch_xla_models/tests/test_spmd.py

```diff
@@ -135,6 +137,210 @@ def test_llama_config_sharding_against_fsdp_v2(self):
       model_config_sharded, model_fsdp_v2_sharded, input, labels
     )
 
+  def test_llama_confg_sharding_against_fsdp_cp(self):
+    import numpy as np
+    import torch_xla.runtime as xr
+    from torch_xla.distributed.spmd import Mesh
+
+    # TODO(https://github.com/pytorch/xla/issues/8063): `xla_force_host_platform_device_count` doesn't
+    # work on PyTorch/XLA. We must run this on the TPU for now.
+    if xr.device_type() != "TPU":
```

Comment:
I'd suggest adding something like this in test utils, we use it quite often:
def skip_unless_tpu(test_func):
    return pytest.mark.skipif(
        xr.device_type() != "TPU",
        reason="This test only works on TPU"
    )(test_func)